src/network/os1/client/prot/game/GameClientRepository.ts

```diff
@@ -30,5 +82,31 @@ export default class GameClientRepository extends ClientRepository {
         this.bind(new ClientCheatDecoder(), new ClientCheatHandler())
         this.bind(new IfButtonDecoder(), new IfButtonHandler())
         this.bind(new ClanDecoder(), new ClanHandler())
+        this.bind(new OpPlayer1Decoder(), new OpPlayer1Handler())
```

Comment:
I would make all of these into single decoder/handlers per *type* (OpPlayer, OpNpc, OpLoc, OpPlayerU, OpNpcU, OpPlayerT, ...), you can pass the opcode into the decoder and track "option" / "op" in the message.

I can give an example on how to implement if you need. Your message model becomes:
```ts
class OpPlayer {
    op: number;
    idx: number;
}
```
And you use that op in the handler.

Reply:
I think I got it the way you wanted. cleaned up the PR quite a bit so thank you